apps/backend/src/drink-action/drink-action.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from 'nestjs-prisma';
+
+import { CreateDrinkActionDto } from './dto/create-drink-action.dto';
+import { UpdateDrinkActionDto } from './dto/update-drink-action.dto';
+import { DrinkAction } from './entities/drink-action.entity';
+
+@Injectable()
+export class DrinkActionsService {
+  constructor(private prisma: PrismaService) {}
+
+  async create(createDrinkActionDto: CreateDrinkActionDto) {
+    return this.prisma.drinkAction.create({ data: createDrinkActionDto });
+  }
```

Comment:
If the drinkId or the eventId is invalid, we get an internal server error. There should be some error handling for these cases.
e.g. :
```suggestion
  async create(createDrinkActionDto: CreateDrinkActionDto) {
    const { drinkId, eventId } = createDrinkActionDto;

    const [drinkExists, eventExists] = await Promise.all([
      this.prisma.drink.findUnique({ where: { id: drinkId } }),
      this.prisma.event.findUnique({ where: { id: eventId } }),
    ]);

    if (!drinkExists) {
      throw new BadRequestException(`Drink with ID ${drinkId} does not exist.`);
    }

    if (!eventExists) {
      throw new BadRequestException(`Event with ID ${eventId} does not exist.`);
    }

    return this.prisma.drinkAction.create({ data: createDrinkActionDto });
  }
  }
```

but there are other (maybe more efficient ) ways to do it.

Reply:
It's probably more efficient to just attempt the update and catch any exceptions. We might not be able to tell which ID was invalid, but since this API won't be public, it's not essential to give a very detailed error message